File: 6.controllers-services-repositories-middlewares-and-request-context/code/go/14use_case_passing_authentication_data_7.go
func CreateBookHandler(w http.ResponseWriter, r *http.Request) {
    var req CreateBookRequest
    json.NewDecoder(r.Body).Decode(&req)

    // Read the trusted user ID FROM THE CONTEXT, not from req.
    // The auth middleware put it there after verifying the token.
    userID := r.Context().Value("userID").(int)
    role := r.Context().Value("role").(string)

    if role != "admin" && role != "user" {
        http.Error(w, "forbidden", http.StatusForbidden)
        return
    }

    // Persist with the SERVER-VERIFIED owner id ,  never the client's.
    book, _ := bookService.Create(r.Context(), req, userID)
    w.WriteHeader(http.StatusCreated) // 201
    json.NewEncoder(w).Encode(book)
}

File: 6.controllers-services-repositories-middlewares-and-request-context/code/python/14use_case_passing_authentication_data_7.py
def create_book_handler(request):
    req = CreateBookRequest(**request.get_json())

    # Read the trusted user id FROM THE CONTEXT, not from the body.
    # The auth middleware put it there after verifying the token.
    user_id = request.context["user_id"]
    role = request.context["role"]

    if role not in ("admin", "user"):
        return Response("forbidden", status=403)

    # Persist with the SERVER-VERIFIED owner id ,  never the client's.
    book = book_service.create(req, user_id)
    return jsonify(book), 201
